lib/Parser/Scan.cpp

```diff
@@ -1723,28 +1727,35 @@ tokens Scanner<EncodingPolicy>::ScanCore(bool identifyKwds)
                 Assert(chType == _C_DIG || chType == _C_DOT);
                 p = m_pchMinTok;
                 this->RestoreMultiUnits(m_cMinTokMultiUnits);
-                bool likelyInt = true;
-                pchT = FScanNumber(p, &dbl, likelyInt, savedMultiUnits);
+                LikelyType likelyType = LikelyInt;
+                pchT = FScanNumber(p, &dbl, likelyType, savedMultiUnits);
                 if (p == pchT)
                 {
                     this->RestoreMultiUnits(savedMultiUnits);
                     Assert(this->PeekFirst(p, last) != '.');
                     Error(ERRbadNumber);
                 }
                 Assert(!Js::NumberUtilities::IsNan(dbl));
-
+                if (likelyType == LikelyBigInt)
+                {
+                    Assert(m_scriptContext->GetConfig()->IsESBigIntEnabled());
+                    token = tkBigIntCon;
+                    m_ptoken->SetBigInt(this->GetHashTbl()->PidHashNameLen(reinterpret_cast<const char *>(p), reinterpret_cast<const char *>(pchT), (uint32)(pchT - p)));
```

Comment:
These casts might be invalid, depending on the EncodingPolicy. The same Scanner code can run on UTF8 or UTF16 input, and EncodedCharPtr is defined to match the size of the input.